src/main/kotlin/io/meshcloud/dockerosb/service/GenericServiceInstanceService.kt

```diff
@@ -15,8 +15,9 @@ import reactor.core.publisher.Mono
 class GenericServiceInstanceService(
     private val yamlHandler: YamlHandler,
     private val gitHandler: GitHandler,
-    private val catalog: Catalog
+    catalogService: GenericCatalogService
 ) : ServiceInstanceService {
+    private val catalog = catalogService.getCatalogInternal()
```

Comment:
f: Same comment as above.  If you do it like this, this reference to the catalog will only be set when this bean is created.